py/modbuiltins.c

```diff
@@ -440,8 +440,18 @@ STATIC mp_obj_t mp_builtin_round(mp_uint_t n_args, const mp_obj_t *args) {
     mp_int_t num_dig = 0;
     if (n_args > 1) {
         num_dig = mp_obj_get_int(args[1]);
-        if (num_dig > 0) {
-            mp_not_implemented("round(..., N>0)");
+        if (num_dig > 0 && num_dig < 7) {
```

Comment:
Yes, no need to test for < 7. Actually, maybe no need to test for >0, then <0 case should actually automagically work.